stainless_extraction/src/ty.rs

```diff
@@ -80,6 +82,12 @@ impl<'l, 'tcx> BaseExtractor<'l, 'tcx> {
         self.extract_ty(substitutions.type_at(0), txtcx, span)
       }
 
+      // String type
+      TyKind::Adt(adt_def, _) if self.is_string(adt_def) => f.StringType().into(),
+
+      // String slice, erased to a plain String
+      TyKind::Str => f.StringType().into(),
```

Comment:
Maybe match on a `TyKind::Ref(TyKind::Str)` instead?

Reply:
That sounds like a good idea.